### PR TITLE
#161 | Added supportStatus to user details.

### PR DESCRIPTION
**Pull request by ilkerCelimli**
# PULL REQUEST TEMPLATE

## Description

Please provide a brief description of the changes made in this pull request.

## Related Issue

If this pull request is related to any GitHub issue, please mention it here with the appropriate issue number.

## Proposed Changes

Please provide a brief and descriptive title for your pull request, summarizing the changes or features introduced. It
should be in English and clearly indicate the purpose of the pull request. Optionally, you can provide a more detailed
description of what you have done.

- Branches should follow the below `Naming Conventions of Branchs` header in [README.md](https://github.com/afet-yonetim-sistemi/ays-be/edit/main/README.md) File

## Additional Information

If there's any additional information or context that the reviewers should know, please provide it here.

## Checklist

Please check the following before submitting your pull request:

- [x] I have tested my changes locally and they are working as expected.
- [x] The code is formatted according to the project's coding guidelines and style.
- [x] Necessary documentation has been added or existing documentation has been updated.
- [x] Relevant unit tests have been written and included.
- [x] Relevant integration tests have been written and included.
- [x] The code has been reviewed to ensure its quality.
- [x] The code does not contain any issues flagged by SonarLint.
- [x] Default reviewers have been assigned to this pull request.
- [x] Assignees have been added if necessary.
- [x] Labels have been applied if necessary.
- [x] If an issue is opened and this pull request is related to that issue, it should not be associated with any project
  board or milestone. However, if there is no issue, the pull request should be associated with the appropriate project
  board (BE Board, In Review column) and milestone to track the progress.

## Reviewers

The default project maintainers and team members will be automatically assigned as reviewers for this pull request.

### CODE REVIEW

File: src/main/java/com/ays/user/model/dto/response/UsersResponse.java [hunk 1]
```diff
@@ -19,7 +20,7 @@ public class UsersResponse {
     private String lastName;
     private UserRole role;
     private UserStatus status;
-
+    private UserSupportStatus supportStatus;
```
Comment (agitrubard) [inline src/main/java/com/ays/user/model/dto/response/UsersResponse.java:23]: Sadece detay endpointinde gösterilecek denmişti diye hatırlıyorum

File: src/main/java/com/ays/user/model/mapper/UserToUserResponseMapper.java [hunk 2]
```diff
@@ -24,4 +25,6 @@ public interface UserToUserResponseMapper extends BaseMapper<User, UserResponse>
     static UserToUserResponseMapper initialize() {
         return Mappers.getMapper(UserToUserResponseMapper.class);
     }
+    @Mapping(source = "supportStatus",target = "supportStatus")
+    UserResponse map(User user);
```
Comment (agitrubard) [inline src/main/java/com/ays/user/model/mapper/UserToUserResponseMapper.java:29]: Burada neden özel bir tanım yapmamız gerekti?

**Review by agitrubard — CHANGES_REQUESTED** [review]

File: src/main/java/com/ays/user/model/dto/response/UsersResponse.java (hunk 1 above)
Comment (ilkerCelimli) [reply src/main/java/com/ays/user/model/dto/response/UsersResponse.java:23]: Detay diye bir end point bulamadım ben, o feature daha eklenmedi mi acaba ?

File: src/main/java/com/ays/user/model/mapper/UserToUserResponseMapper.java (hunk 2 above)
Comment (ilkerCelimli) [reply src/main/java/com/ays/user/model/mapper/UserToUserResponseMapper.java:29]: Normal mappinde SuppotStatus null geliyordu,Herhangi bir mapping işlemi yapılmadığı için mecburen oraya özel tanım yapılması gerekildi.

**Review by agitrubard — APPROVED** [review]

**Review by moaydogdu — APPROVED** [review]
LGTM!

**Pull request merged**